recipes/dcmtk/all/conanfile.py

```diff
@@ -167,147 +170,116 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "etc"))
         tools.rmdir(os.path.join(self.package_folder, "share"))
 
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._module_file_rel_path),
+            {target: "DCMTK::{}".format(target) for target in self._dcmtk_components.keys()}
+        )
+
+    @staticmethod
+    def _create_cmake_module_alias_targets(module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        tools.save(module_file, content)
+
+    @property
+    def _module_subfolder(self):
+        return os.path.join("lib", "cmake")
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join(self._module_subfolder,
+                            "conan-official-{}-targets.cmake".format(self.name))
+
+    @property
+    def _dcmtk_components(self):
+        def charset_conversion():
+            if bool(self.options.charset_conversion):
+                return ["libiconv::libiconv"] if self.options.charset_conversion == "libiconv" else ["icu::icu"]
+            return []
+
+        def zlib():
+            return ["zlib::zlib"] if self.options.with_zlib else []
+
+        def png():
+            return ["libpng::libpng"] if self.options.with_libpng else []
+
+        def tiff():
+            return ["libtiff::libtiff"] if self.options.with_libtiff else []
+
+        def openssl():
+            return ["openssl::openssl"] if self.options.with_openssl else []
+
+        def tcpwrappers():
+            return ["tcp-wrappers::tcp-wrappers"] if self.options.get_safe("with_tcpwrappers") else []
+
+        def xml2():
+            return ["libxml2::libxml2"] if self.options.with_libxml2 else []
+
+        return {
+            "ofstd"   : charset_conversion(),
+            "oflog"   : ["ofstd"],
+            "dcmdata" : ["ofstd", "oflog"] + zlib(),
+            "i2d"     : ["dcmdata"],
+            "dcmimgle": ["ofstd", "oflog", "dcmdata"],
+            "dcmimage": ["oflog", "dcmdata", "dcmimgle"] + png() + tiff(),
+            "dcmjpeg" : ["ofstd", "oflog", "dcmdata", "dcmimgle", "dcmimage", "ijg8", "ijg12", "ijg16"],
+            "ijg8"    : [],
+            "ijg12"   : [],
+            "ijg16"   : [],
```

Comment:
should the order be flipped?

Reply:
What do you mean?

Reply:
It seems odd to require components which have no been created, perhaps I am not following this correctly?

Reply:
Order of creation of conan components doesn't matter. Moreover here it's a dictionnary, there is no order.